packages/SwingSet/src/kernel/kernel.js

```diff
@@ -597,7 +597,7 @@ export default function buildKernel(
     const p = kernelKeeper.getKernelPromise(kpid);
     kernelKeeper.incStat('dispatchNotify');
     const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
-    p.state !== 'unresolved' || Fail`spurious notification ${kpid}`;
+    assert(p.state !== 'unresolved', `spurious notification ${kpid}`);
```

Comment:
FYI, if this was in a hot path, this would be a perf regression. It's unfortunate that TS is not capable of doing flow analysis in this case. The faster but more verbose alternative is to do

```suggestion
    if (p.state !== 'unresolved') {
      throw Fail`spurious notification ${kpid}`;
    }
```

Reply:
It's `if (p.state === 'unresolved')`, but I'll do it.